src/common/hooks/use-setup-tx.ts

```diff
@@ -32,10 +33,23 @@ export const useSetupTx = () => {
   }, [requestToken, handleNetworkSwitch, handleAccountSwitch]);
 
   useEffect(() => {
-    if (currentAccountStxAddress) {
-      if (!previousAccountStxAddress || previousAccountStxAddress !== currentAccountStxAddress) {
+    if (requestToken && currentAccountStxAddress) {
+      if (
+        !hasMounted ||
+        !previousAccountStxAddress ||
+        previousAccountStxAddress !== currentAccountStxAddress
+      ) {
+        if (!hasMounted) {
```

Comment:
Maybe I'm not reading this right, but why does this condition need to be nested here (checked twice)?

Reply:
Nvm, came back and looked closer. I see it.

Reply:
yeah i mean it is confusing lol, it's basically saying this:

```ts
// if we the token has loaded, and all the side effects from our account logic is ready
    if (requestToken && currentAccountStxAddress) {
      if (
      // if this is false, that means it's the first time it has run, and we want to run it at least 1 time
        !hasMounted ||
        !previousAccountStxAddress || // tbh this line might not need to be here, it might be for the next line
        
        // this means the account has been switched by the user, and we need to make sure the post conditions have been updated to reflect this change
        previousAccountStxAddress !== currentAccountStxAddress
      ) {
        if (!hasMounted) {
          setHasMounted(true);
        }
        void handlePostConditions(currentAccountStxAddress);
      }
    }
    ```

Reply:
Thanks for the feedback. I appreciate your code comments. 👍